RIOT-OS/RIOT
# Add driver for SPS30 particulate matter sensor

### Contribution description
This PR provides a sensor driver for the [Sensirion SPS30](https://www.sensirion.com/en/environmental-sensors/particulate-matter-sensors-pm25/) particulate matter sensor.
~I added the WIP label for now because I'm still planning to add SAUL support and improve the test application a bit.~ *[edit:]* done
For now I only implemented I2C support but the sensor could also be used via UART.
Any opinions on whether I should add that too? If yes, how (i.e. runtime vs. compiletime selection) ?




### Testing procedure
Check the documentation of `sps30.h` to see if it is sufficient to get it running and run the test application under `tests/driver_sps30`.
*[edit:]*
SAUL adaption is now included so the generic saul example can also be used for testing:
`USEMODULE="sps30" BOARD=<your_board_name> make -C examples/saul all flash term`

*[edit 2:]*
Keep in mind that the I2C bus must be running in standard mode speed. You might need to change that in your boards `periph_conf.h`




### Issues/PRs references
Might be a good candidate for discussing about drivers with support for multiple interfaces (i.e. #13208).
